179 unable to logout when two browser windows are open

steven-chiu-bigstack opened the pull request:
#### What type of PR is this?

Fix

#### What this PR does / why we need it

Fix unable to logout when two (or more) COS windows are opened bug.

#### Which issue(s) this PR fixes

Fixes #179 

#### Special notes for your reviewer

In short, I replace all `useCosStreamRequest` with `useCosGetRequest` + `useSequentialInterval` for polling. 

- Single request:

https://github.com/user-attachments/assets/4120fb86-41b5-4cb2-b85d-a0fa86aa8138

- Multiple requests:
 
https://github.com/user-attachments/assets/81a64e45-252f-4a0c-8f88-1251d133f772

steven-chiu-bigstack committed:
refactor(web-app): replace stream request with polling in Home Overview Page

steven-chiu-bigstack commented on packages/cube-frontend-web-app/src/pages/home/overview/_components/HealthPanel/HealthPanel.tsx:
```diff
@@ -61,12 +68,14 @@ const HealthPanel = () => {
+          onRepair={handleRepair}
+          isRepairButtonLoading={isRepairButtonLoading}
+        />
+      )}
```
Added a check to only show `HealthError` when there are errors:

![image](https://github.com/user-attachments/assets/585d67fa-d782-47e4-ad0a-32e22183dc3e)

steven-chiu-bigstack committed:
refactor(web-app): replace stream request with polling in Home Health Page

Signed-off-by: steven-chiu-bigstack <steven.chiu@bigstack.co>

steven-chiu-bigstack committed:
fix(web-app): home overview page health panel error section display

Signed-off-by: steven-chiu-bigstack <steven.chiu@bigstack.co>

raven-pan reviewed (approved):
LGTM, Thanks for your help in solving this issue!

I noticed that the current `useInterval` strategy results in a lot of pending requests.

We can handle this in issue https://github.com/bigstack-oss/cube-cos-ui/issues/189.

https://github.com/user-attachments/assets/8146a4e6-4586-4da2-9891-5a49e4ff6135

steven-chiu-bigstack committed:
refactor(web-app): replace useInterval with useSequentialInterval for polling

steven-chiu-bigstack commented:
Hi @raven-pan 

Thanks for pointing out the pending requests problem! I've updated the PR that introduced the `useSequentialInterval` hook and replaced all the polling `useInterval` with it.
Please have a second look when you have time, thank you! 🙏

steven-chiu-bigstack committed:
fix(web-app): home health bar charts constantly showing skeletons

Signed-off-by: steven-chiu-bigstack <steven.chiu@bigstack.co>

steven-chiu-bigstack committed:
fix(web-app): display logic when there's only 1 item in the history of health segmented bar

Signed-off-by: steven-chiu-bigstack <steven.chiu@bigstack.co>
